src/logic/map_objects/tribes/ship.cc

```diff
@@ -1143,51 +1144,57 @@ const Bob::Task* Ship::Loader::get_task(const std::string& name) {
 	return Bob::Loader::get_task(name);
 }
 
-void Ship::Loader::load(FileRead& fr, uint8_t /* packet_version */) {
-	Bob::Loader::load(fr);
-	// Economy
-	ware_economy_serial_ = fr.unsigned_32();
-	worker_economy_serial_ = fr.unsigned_32();
-
-	// The state the ship is in
-	ship_state_ = static_cast<ShipStates>(fr.unsigned_8());
-
-	// Expedition specific data
-	if (ship_state_ == ShipStates::kExpeditionScouting ||
-	    ship_state_ == ShipStates::kExpeditionWaiting ||
-	    ship_state_ == ShipStates::kExpeditionPortspaceFound ||
-	    ship_state_ == ShipStates::kExpeditionColonizing) {
-		expedition_.reset(new Expedition());
-		// Currently seen port build spaces
-		expedition_->seen_port_buildspaces.clear();
-		uint8_t numofports = fr.unsigned_8();
-		for (uint8_t i = 0; i < numofports; ++i) {
-			expedition_->seen_port_buildspaces.push_back(read_coords_32(&fr));
-		}
-		// Swimability of the directions
-		for (uint8_t i = 0; i < LAST_DIRECTION; ++i) {
-			expedition_->swimmable[i] = (fr.unsigned_8() == 1);
+void Ship::Loader::load(FileRead& fr, uint8_t packet_version) {
+	if (packet_version == kCurrentPacketVersionMapObject) {
```

Comment:
`packet_version` here refers to the _ship_ packet version. There is no need to compare `kCurrentPacketVersionMapObject` with anything here as far as I can see.

Reply:
Yep, that's a bug, and it broke the test suite too. Fixed.